compiler/rustc_passes/src/liveness.rs

```diff
@@ -993,11 +1008,26 @@ impl<'a, 'tcx> Liveness<'a, 'tcx> {
 
             hir::ExprKind::MethodCall(.., ref args, _) => {
                 let m = self.ir.tcx.parent_module(expr.hir_id).to_def_id();
-                let succ = if self.ir.tcx.is_ty_uninhabited_from(
-                    m,
-                    self.typeck_results.expr_ty(expr),
-                    self.param_env,
-                ) {
+                let ty = self.typeck_results.expr_ty(expr);
+                let succ = if self.ir.tcx.is_ty_uninhabited_from(m, ty, self.param_env) {
+                    if let LiveNodeKind::ExprNode(succ_span, succ_id) = self.ir.lnks[succ] {
```

Comment:
Same as above, and given this is the second time you're doing this, maybe adding a method to `LiveNodeKind` might be warranted?